src/midi/midi_mt32_model.cpp

```diff
@@ -0,0 +1,145 @@
+	assert(ctrlFull || (ctrlA && ctrlB));
+}
+
+// Checks if its ROMs can be positively found in the provided directory
+bool Model::InDir(const service_t &service, const std::string &dir)
+{
+	assert(service);
+	if (inDir.find(dir) != inDir.end())
+		return inDir[dir];
+
+	auto find_rom = [&service, &dir](const Rom *Rom) -> bool {
+		if (!Rom)
+			return false;
+
+		const std::string rom_path = dir + Rom->filename;
+		if (!path_exists(rom_path))
+			return false;
+
+		mt32emu_rom_info info;
+		if (service->identifyROMFile(&info, rom_path.c_str(), nullptr) != MT32EMU_RC_OK)
+			return false;
+
+		if (Rom->type == ROM_TYPE::UNVERSIONED)
+			return true;
+
+		const bool found_pcm = info.pcm_rom_id && (Rom->id == info.pcm_rom_id);
+		const bool found_ctrl = info.control_rom_id && (Rom->id == info.control_rom_id);
+		return found_pcm || found_ctrl;
+	};
+
+	auto find_both = [find_rom](const Rom *rom_a, const Rom *rom_b) -> bool {
+		return find_rom(rom_a) && find_rom(rom_b);
+	};
+
+	const bool have_pcm = find_rom(pcmFull) || find_both(pcmA, pcmB);
+	const bool have_ctrl = find_rom(ctrlFull) || find_both(ctrlA, ctrlB);
+	const bool have_both = have_pcm && have_ctrl;
+	inDir[dir] = have_both;
+	return have_both;
+}
+
+// If present, loads either the full or partial ROMs from the provided directory
+bool Model::Load(const service_t &service, const std::string &dir)
+{
+	if (!service || !InDir(service, dir))
+		return false;
+
+	auto load_rom = [&service, &dir](const Rom *rom_full,
+	                                 mt32emu_return_code expected_code) -> bool {
+		if (!rom_full)
+			return false;
+		const std::string rom_path = dir + rom_full->filename;
+		const auto rcode = service->addROMFile(rom_path.c_str());
+		return rcode == expected_code;
+	};
+	auto load_both = [&service, &dir](const Rom *rom_a, const Rom *rom_b,
+	                                  mt32emu_return_code expected_code) -> bool {
+		if (!rom_a || !rom_b)
+			return false;
+		const std::string rom_a_path = dir + rom_a->filename;
+		const std::string rom_b_path = dir + rom_b->filename;
+		const auto rcode = service->mergeAndAddROMFiles(rom_a_path.c_str(),
+		                                                rom_b_path.c_str());
+		return rcode == expected_code;
+	};
+	const bool loaded_pcm = load_rom(pcmFull, MT32EMU_RC_ADDED_PCM_ROM) ||
+	                        load_both(pcmA, pcmB, MT32EMU_RC_ADDED_PCM_ROM);
+	const bool loaded_ctrl = load_rom(ctrlFull, MT32EMU_RC_ADDED_CONTROL_ROM) ||
+	                         load_both(ctrlA, ctrlB, MT32EMU_RC_ADDED_CONTROL_ROM);
+	return loaded_pcm && loaded_ctrl;
+}
+
+// Returns the model's version, which is postfixed on it's name. 
+// If a version doesn't exist, 
+const std::string &Model::Version()
+{
+	if (!version.empty())
+		return version;
+
+	const auto pos = name.find_first_of('_');
+	version = (pos == std::string::npos) ? name : name.substr(pos + 1);
+	return version;
+}
+const std::string &Model::Name() const
```

Comment:
1. Add an empty line to separate from the method above, please.
2. Never return fields via a reference.
3. Implementation is trivial, perhaps should be in header. (Also, elsewhere I suggested returnin `const char*` instead of `std::string` and renaming.